workspaces/bi/bi-extension/src/test/e2e-playwright-tests/type/type.spec.ts

```diff
@@ -80,6 +80,56 @@ export default function createTests() {
             await typeUtils.saveAndWait(editForm);
             await typeUtils.verifyTypeLink(recordName, 'role', enumName);
 
+            // Add name field to Employee record
+            await typeUtils.editType(recordName);
+            await typeUtils.addRecordField('name', 'string' );
+
+            // Toggle drop down
+            await typeUtils.toggleFieldOptionsByChevron(2);
+            await typeUtils.setCheckbox('Readonly', true);
+            await typeUtils.verifyCheckboxStates(
+                {
+                    'Readonly': true,
+                }
+            );
+
+        
+            // Test Advanced Options functionality
+            console.log('Expanding Advanced Options...');
+            await typeUtils.toggleDropdown('Advanced Options');
+        
+            console.log('Testing Allow Additional Fields checkbox...');
+            await typeUtils.setCheckbox('Allow Additional Fields', true);
+            await typeUtils.verifyCheckboxStates(
+                {
+                    'Allow Additional Fields': true,
+                    'Is Readonly Type': false
+                }
+  
+            );
+        
+            
+            console.log('Testing Is Readonly Type checkbox...');
+            await typeUtils.setCheckbox('Is Readonly Type', true);
+            await typeUtils.verifyCheckboxStates(
+                {
+                    'Allow Additional Fields': true,
+                    'Is Readonly Type': true
+                }
+            );
+        
+            console.log('Testing unchecking Allow Additional Fields...');
+            await typeUtils.setCheckbox('Allow Additional Fields', false);
+            await typeUtils.verifyCheckboxStates(
+                {
+                    'Allow Additional Fields': false,
+                    'Is Readonly Type': true
+                }
+            );
```

Comment:
I think it would be better to verify whether the generated source updates correctly based on the checkbox interaction, rather than checking the checkbox states themselves